monai/bundle/workflows.py

```diff
@@ -0,0 +1,681 @@
+                parser[k] = v
+        # save the executed config into file
+        default_name = f"config_{time.strftime('%Y%m%d_%H%M%S')}.json"
+        filepath = parser.get("execute_config", None)
+        if filepath is None:
+            if "output_dir" not in parser:
+                # if no "output_dir" in the bundle config, default to "<bundle root>/eval"
+                parser["output_dir"] = "$@bundle_root + '/eval'"
+            # experiment management tools can refer to this config item to track the config info
+            parser["execute_config"] = parser["output_dir"] + f" + '/{default_name}'"
+            filepath = os.path.join(parser.get_parsed_content("output_dir"), default_name)
+        Path(filepath).parent.mkdir(parents=True, exist_ok=True)
+        parser.export_config_file(parser.get(), filepath)
+
+
+class ConfigTrainWorkflow(ConfigWorkflow, TrainProperties):
+    required_train_ids = [
+        "bundle_root",
+        "device",
+        "dataset_dir",
+        "train#trainer",
+        "train#trainer#max_epochs",
+        "train#dataset",
+        "train#dataset#data",
+        "train#handlers",
+        "train#inferer",
+    ]
+    required_val_ids = [
+        "validate#evaluator",
+        "validate#handlers",
+        "validate#dataset",
+        "validate#dataset#data",
+        "validate#inferer",
+    ]
+    optional_train_ids = ["train#preprocessing", "train#postprocessing", "train#key_metric"]
+    optional_val_ids = ["validate#preprocessing", "validate#postprocessing", "validate#key_metric"]
+
+    def check(self) -> bool:
+        ret = self._check_required_ids(self.required_train_ids)
+        # if having validation logic, verify the ids
+        ret &= self._check_required_ids(self.required_val_ids) if "validate" in self.parser else True
+        # check optional ids if existing
+        ret &= (
+            self._check_optional_id("train#dataset#transform", self.optional_train_ids[0])
+            & self._check_optional_id("train#trainer#postprocessing", self.optional_train_ids[1])
+            & self._check_optional_id("train#trainer#key_train_metric", self.optional_train_ids[2])
+            & self._check_optional_id("validate#dataset#transform", self.optional_val_ids[0])
+            & self._check_optional_id("validate#evaluator#postprocessing", self.optional_val_ids[1])
+            & self._check_optional_id("validate#evaluator#key_val_metric", self.optional_val_ids[2])
+        )
+        for h in self.parser["train#handlers"]:
+            if h["_target_"] == "ValidationHandler":
+                interval = h.get("interval", None)
+                if interval is not None and interval != "val_interval":
+                    logger.info(f"please use id 'val_interval' to define validation interval, got: '{interval}'.")
+                    ret = False
+        return ret
+
+    @property
+    def bundle_root(self) -> str:
+        return self.get_content(self.required_train_ids[0])
```

Comment:
Hi @Nic-Ma , here can `required_train_ids` and `required_val_ids` be a dict? If changing the list, using the index to get the key may return a wrong result.

Reply:
or maybe we can define enum classes

Reply:
`Enum` may be a good idea, let me try it later.

Thanks.